viewer/packages/sector-parser/src/GltfSectorParser.ts

```diff
@@ -2,224 +2,289 @@
+          return 'a_treeIndex';
+        default:
+          throw new Error();
+      }
+    }
   }
 
-  private processInstancedMesh(
-    mesh: Mesh | null,
-    instanced: InstancedMesh | null,
+  private setIndexBuffer(
+    payload: GeometryProcessingPayload,
+    primitive: Primitive,
+    data: ArrayBuffer,
     bufferGeometry: THREE.InstancedBufferGeometry | THREE.BufferGeometry
   ) {
-    assert(mesh !== null, 'Instanced mesh node does not contain a mesh');
-    assert(instanced !== null, 'Instanced mesh does not contain the instanced gltf extension');
+    const json = payload.glbHeaderData.json;
 
-    const primitive = mesh.listPrimitives()[0];
-    this.setMeshAttributes(primitive, bufferGeometry);
+    const offsetToBinChunk = payload.glbHeaderData.byteOffsetToBinContent;
 
-    const instanceId = mesh.getExtras()['InstanceId'] as number;
+    const indicesAccessor = json.accessors[primitive.indices];
+    const indicesBufferView = json.bufferViews[indicesAccessor.bufferView];
+    indicesBufferView.byteOffset = indicesBufferView.byteOffset ?? 0;
 
-    assert(instanceId !== undefined, 'Instanced mesh does not have an instance ID');
+    const IndicesTypedArrayConstructor = this.DATA_TYPE_BYTE_SIZES.get(indicesAccessor.componentType)!;
 
-    this.setInstancedAttributes(instanced, bufferGeometry);
+    const indicesTypedArray = new IndicesTypedArrayConstructor(
+      data,
+      offsetToBinChunk + indicesBufferView.byteOffset,
+      indicesBufferView.byteLength / IndicesTypedArrayConstructor.BYTES_PER_ELEMENT
+    );
 
-    return instanceId;
-  }
+    const elementSize = this.COLLECTION_TYPE_SIZES.get(indicesAccessor.type)!;
 
-  private isEmptyNode(instanced: InstancedMesh | null, mesh: Mesh | null) {
-    return !instanced && !mesh;
+    bufferGeometry.setIndex(new THREE.BufferAttribute(indicesTypedArray, elementSize));
   }
 
-  private setInstancedAttributes(
-    instanced: InstancedMesh,
-    bufferGeometry: THREE.BufferGeometry | THREE.InstancedBufferGeometry
+  private setInterleavedBufferAttributes<T extends THREE.InterleavedBuffer>(
+    glbHeaderData: GlbHeaderData,
+    attributes: {
+      [key: string]: number;
+    },
+    data: ArrayBuffer,
+    transformAttributeName: (attributeName: string) => string,
+    bufferGeometry: THREE.BufferGeometry | THREE.InstancedBufferGeometry,
+    bufferType: { new (array: ArrayLike<number>, stride: number): T }
   ) {
-    instanced.listSemantics().forEach(attributeName => {
-      const accessor = instanced!.getAttribute(attributeName)!;
-      const instancedBufferAttribute = new THREE.InstancedBufferAttribute(
-        accessor.getArray()!,
-        accessor.getElementSize()
-      );
+    const json = glbHeaderData.json;
 
-      bufferGeometry.setAttribute(`a${attributeName}`, instancedBufferAttribute);
-    });
-  }
+    const uniqueBufferViews = [
+      ...new Set(Object.values(attributes).map(accessorId => json.accessors[accessorId].bufferView))
+    ];
```

Comment:
Not 100% sure on whats going on here, but isn't it a bit excessive to create a Set if theres only one element?

Reply:
Yeah, wasn't 100% satisfied with this. What I am trying to do is get all unique values of a given list of numbers. Basically want Linq `Distinct` method here. Any suggestions?

Reply:
From what I can understand, `Object.values()` returns a list. It is funneled into a `Set` for deduplication

Reply:
All you are trying to do is to ensure that all values are equal right? In this case you'd only need to iterate through them and ensure they are all equal to the first element.